### ISSUE TITLE
[New Check] uselessly greedy parameter expansion for prefix without unquoted *

### ISSUE THREAD

**Issue by guillaume-d** [koalaman/shellcheck#2507]
#### For new checks and feature suggestions

- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

I searched for: "parameter expansion", "prefix", "suffix", "shortest", "longest", "greedy", "lazy".
Non-alphanumeric-only search ("##", "%%") does not seem to work in GitHub issues.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
# using 'l' and 'r' for easier reading, but usually whitespace is to be trimmed:
untrimmed=llABCrr
echo "left-trimmed: ${untrimmed##l}"
echo "right-trimmed: ${untrimmed%%r}"
```

#### Here's what shellcheck currently says:

```
No issues detected!
```

#### Here's what I wanted or expected to see:

```
script.sh:4: needlessly greedy parameter expansion for prefix without unquoted *
script.sh:5: needlessly greedy parameter expansion for suffix without unquoted *
```

Also a "Did you mean:" note would be great for the common stripping problem above, which
I as a novice probably was not alone in getting wrong: I though doubling the operator would strip as much as possible.
This problem generalizes to any prefix or suffix being a [glob](https://manned.org/man/glob.7) character class (AKA [POSIX's RE Bracket Expression](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html#tag_09_03_05])): AFAICU it can always be easily (syntactically) inverted!
Likewise in the new check's wiki page one `fork`-free, POSIX-shell-only solution for this trimming problem could be added (_not my idea_, see for example https://github.com/dylanaraps/pure-sh-bible#trim-leading-and-trailing-white-space-from-string):

```
$ untrimmed='llArr'
$ echo "${untrimmed##l}" # not working as expected
lArr
$ echo "${untrimmed%%r}" # not working as expected
llAr
$ left=${untrimmed%%[^l]*}; echo "${untrimmed#"$left"}" # OK
Arr
$ right=${untrimmed##*[^r]}; echo "${untrimmed%"$right"}" # OK
llA
```
